packages/core-utils/src/app/contract-deployment.ts

```diff
@@ -109,7 +109,7 @@ export const deploy = async (
     await sleep(5_000)
   }
 
-  return deployContractFunction(wallet)
+  return deployContractFunction(wallet, provider)
```

Comment:
We can certainly make it explicit, but note, wallet makes the provider accessible at `wallet.provider`. See [line 103](https://github.com/ethereum-optimism/optimism-monorepo/pull/149/files#diff-ab2c8669acb72c7e2c7aec733a1f4092R103)

Reply:
updated to use `wallet.provider` instead!